src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+    using System.Xml;
+    using Microsoft.OData.Buffers;
+    using Microsoft.OData.Edm;
+    #endregion Namespaces
+
+    /// <summary>
+    /// Provides helper method for converting data values to and from the OData JSON format.
+    /// </summary>
+    internal static partial class JsonValueUtils
+    {
+        /// <summary>
+        /// Write a char value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">The char value to write.</param>
+        /// <param name="stringEscapeOption">The ODataStringEscapeOption to use in escaping the string.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, char value, ODataStringEscapeOption stringEscapeOption)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            if (stringEscapeOption == ODataStringEscapeOption.EscapeNonAscii || value <= 0x7F)
+            {
+                string escapedString = SpecialCharToEscapedStringMap[value];
+                if (escapedString != null)
+                {
+                    await writer.WriteAsync(escapedString).ConfigureAwait(false);
+                    return;
+                }
+            }
+
+            await writer.WriteAsync(value).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a boolean value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">The boolean value to write.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, bool value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value ? JsonConstants.JsonTrueLiteral : JsonConstants.JsonFalseLiteral).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write an integer value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Integer value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, int value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a float value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Float value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, float value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            if (float.IsInfinity(value) || float.IsNaN(value))
+            {
+                await writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo)).ConfigureAwait(false);

```

Comment:
Add Culture info in the `ToString`

Reply:
@KenitoInc
The `.ToString()` overloads available when converting a number to string are as follows:
- `.ToString()`
- `.ToString(string format)`
- `.ToString(IFormatProvider provider)`
- `.ToString(string format, IFormatProvider provider)`

We're using the 3rd overload by supplying the `IFormatProvider` - `ODataNumberFormatInfo`. There's no overload that takes `CultureInfo`. That is usually applicable when using `string.Format()`